src/nci_adapter.c

```diff
@@ -397,6 +521,46 @@ nci_adapter_create_known_tag(
     return NULL;
 }
 
+static
+NfcTag*
+nci_adapter_create_other_tag(
+    NciAdapter* self,
+    const NciIntfActivationNtf* ntf)
```

Comment:
I would prefer it this way:
```c
static
const NfcParamPoll*
nci_adapter_get_mode_param(
    NfcParamPoll* poll,
    const NciIntfActivationNtf* ntf)
```
and then the caller would pass that to `nfc_adapter_add_other_tag2()`.

Reply:
Actually, it's pretty much a matter of taste, you can keep it as it if you like. But in this case don't return any value (since it's not being used) and call `NFC_ADAPTER(self)` once at the beginning of the function. That would do it.

Reply:
Thanks, I like your approach, lets try again, I have refactored a little bit